### PR TITLE
Orthogonal C1

### PR DESCRIPTION
**Pull request by TristanPfersdorff**
All the stuff to finish C1 for case O, including the construction for the isotropic subspace stabilizers and the generic function.

### General
- [ ] All functions have unit tests

### Functions constructing generators of maximal subgroups
- [ ] The code which computes and then stores the sizes is correct. To do this one confers to the tables referenced by the code.
- [ ] The test suite checks whether all constructed generators are sensible. That is it tests the generators'
  - [ ] determinants (and if applicable also the spinor norms are correct), and
  - [ ] compatibility with the correct form,
  - [ ] `DefaultFieldOfMatrixGroup` returns the correct field.

### Functions assembling the list of all maximal subgroups of a certain group
- [ ] The code agrees with the tables in section 8.2 of the book "The Maximal Subgroups of the Low-dimensional Finite Classical Groups".

The reviewer doesn't need to compare our results to magma's results. That's the job of the person implementing the code.

### CODE REVIEW

File: gap/Utils.gd [hunk 1]
```diff
@@ -22,7 +22,15 @@ DeclareGlobalFunction("MatrixByEntries");
 #! - If <A>entries</A> is an integer, the entries of <C>M</C> are all ones 
 #!   and the number of them is <A>entries</A>.
 DeclareGlobalFunction("AntidiagonalMat");
-
+#! @Chater Utility Functions
```
Comment (fingolfin) [inline gap/Utils.gd:25]:
```suggestion
#! @Chapter Utility Functions
```

File: tst/standard/ReducibleMatrixGroups.tst [hunk 2]
```diff
@@ -74,6 +74,26 @@ Error, <d> must be even
 gap> SpStabilizerOfNonDegenerateSubspace(4, 2, 3);
 Error, <k> must be less than <d> / 2
 
+#
+gap> TestOmegaStabilizerOfIsotropicSubspace := function(epsilon, d, q, k)
+>   local G;
+>   G := OmegaStabilizerOfIsotropicSubspace(epsilon, d, q, k);
+>   Assert(0, CheckSize(G));
+>   Assert(0, IsSubsetOmega(epsilon, d, q, G));
```
Comment (fingolfin) [inline tst/standard/ReducibleMatrixGroups.tst:82]: I recommend running CheckSize last (see also PR #108; note that this PR effectively conflicts with your PR, as I renamed `IsSubsetOmega` and changed how it is used, see there)

File: tst/standard/ReducibleMatrixGroups.tst [hunk 3]
```diff
@@ -74,6 +74,39 @@ Error, <d> must be even
 gap> SpStabilizerOfNonDegenerateSubspace(4, 2, 3);
 Error, <k> must be less than <d> / 2
 
+#
+gap> TestOmegaStabilizerOfIsotropicSubspace := function(epsilon, d, q, k)
+>   local G;
+>   G := OmegaStabilizerOfIsotropicSubspace(epsilon, d, q, k);
+>   Assert(0, IsSubsetOmega(epsilon, d, q, G));
+>   Assert(0, CheckSize(G));
```
Comment (fingolfin) [inline tst/standard/ReducibleMatrixGroups.tst:82]:
```suggestion
>   CheckIsSubsetOmega(epsilon, d, q, G);
>   CheckSize(G);
```
to match changes introduced in PR #108

**Comment by fingolfin** [comment]
@TristanPfersdorff Is this ready for merging now, from your POV? Or do you plan further changes?

**Comment by TristanPfersdorff** [comment]
Well, I have been working on a nice function `AlternativeGeneratorsOfOrthogonalGroup` in my C2-PR which abstracts away some of the slightly ugly code in this PR (the stuff with the discriminants in the NonDegenerate case with epsilon = 0) by constructing nice generators for the diagonal forms in case q odd, so I do plan to replace the ugly code by that function at some point down the line. But I don't think that is a good reason to block this PR, so I'd say this PR is ready for merging.

**Pull request merged**